CMakeLists.txt

```diff
@@ -139,6 +139,16 @@ set(_SHARED_LIB_SUFFIX
 set(_SUNDIALS_ALIAS_TARGETS ""
     CACHE INTERNAL "" FORCE)
 
+# We default to release builds
+set(_DEFAULT_CMAKE_BUILD_TYPE Release)
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Building SUNDIALS in '${_DEFAULT_CMAKE_BUILD_TYPE}' mode as CMAKE_BUILD_TYPE was not specified.")
+  set(CMAKE_BUILD_TYPE "${_DEFAULT_CMAKE_BUILD_TYPE}" CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+else()
+  message(STATUS "Building SUNDIALS in '${CMAKE_BUILD_TYPE}' mode.")
+endif()
```

Comment:
I don't think so, since with the multi-config generators the user selects a configuration at build time. `CMAKE_CONFIGURATION_TYPES` holds all of the available configuration types, not just the 'selected' one.

Reply:
Ah, yes. I missed that the docs say _available_ build types.